lib/portage_utils.py

```diff
@@ -152,38 +160,42 @@ def open(filename, mode='r', quiet=True):
    filename: name of the file to open
    mode: open mode
    quiet:  suppress "zcat: stdout: Broken pipe" messages.
+   encoding: defaults to "utf8" for text modes
    return: file handle to the open file.
    """
+   if encoding is None and mode not in ("rb", "wb"):
```

Comment:
If the goal is to set the encoding when we are in binary mode, shouldn't we do
```
if encoding is None and 'b' not in mode:
```
instead to catch `w+`, `w+b`, `r+` and `r+b`?

Reply:
I made some revisions to this file.

Now the default is `encoding="utf8"` in the function signature.

How about
```
if 'b' not in mode:
    encoding=None
```
to catch all the possible binary cases?